server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -272,61 +271,35 @@ protected void doStart() {
     }
 
     @SuppressWarnings("this-escape")
-    public IndicesService(
-        Settings settings,
-        PluginsService pluginsService,
-        NodeEnvironment nodeEnv,
-        NamedXContentRegistry xContentRegistry,
-        AnalysisRegistry analysisRegistry,
-        IndexNameExpressionResolver indexNameExpressionResolver,
-        MapperRegistry mapperRegistry,
-        NamedWriteableRegistry namedWriteableRegistry,
-        ThreadPool threadPool,
-        IndexScopedSettings indexScopedSettings,
-        CircuitBreakerService circuitBreakerService,
-        BigArrays bigArrays,
-        ScriptService scriptService,
-        ClusterService clusterService,
-        Client client,
-        FeatureService featureService,
-        MetaStateService metaStateService,
-        Collection<Function<IndexSettings, Optional<EngineFactory>>> engineFactoryProviders,
-        Map<String, IndexStorePlugin.DirectoryFactory> directoryFactories,
-        ValuesSourceRegistry valuesSourceRegistry,
-        Map<String, IndexStorePlugin.RecoveryStateFactory> recoveryStateFactories,
-        List<IndexStorePlugin.IndexFoldersDeletionListener> indexFoldersDeletionListeners,
-        Map<String, IndexStorePlugin.SnapshotCommitSupplier> snapshotCommitSuppliers,
-        CheckedBiConsumer<ShardSearchRequest, StreamOutput, IOException> requestCacheKeyDifferentiator,
-        Supplier<DocumentParsingObserver> documentParsingObserverSupplier
-    ) {
-        this.settings = settings;
-        this.threadPool = threadPool;
-        this.pluginsService = pluginsService;
-        this.nodeEnv = nodeEnv;
+    IndicesService(IndicesServiceBuilder builder) {
```

Comment:
I usually prefer the builder to pass each arg separately to a private/package private ctor, but I can understand why you did it this way: there are so many args.
Maybe a middle ground is to have a data-only class to pass the args, IndicesServicesArgs or similar. It could possibly make clearer the interactions and ordering (e.g. ensure that the ctor is called only when the builder is done - build() is called and IndicesServicesArgs is created and passed), but it's very small, so I'm fine either way.

Reply:
If this was used in more places, then yes I would split it out. But it's only used in node construction and a test, so its not worth the extra class.